File: since2020/CodeForces/1325C.cpp
/**
 *  @judge CodeForces
 *  @id 1325C
 *  @name Ehab and Path-etic MEXs
 *  @contest Codeforces Round #628
 * 
 *  @tag Greedy, Graph Theory
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int LL;
int n;
vector<int> cnt, label;
vector<pair<int, int>> edges;

void solve() {
	int u, v;
	int L = 0;
	for (int i = 0; i < n - 1; i++) {
		tie(u, v) = edges[i];
		if (cnt[u] == 1) {
			label[i] = L;
			L++;
		}
		else if (cnt[v] == 1) {
			label[i] = L;
			L++;
		}
	}
	for (int i = 0; i < n - 1; i++) {
		tie(u, v) = edges[i];
		if (cnt[u] != 1 && cnt[v] != 1) {
			label[i] = L;
			L++;
		}
	}
	return ;
}

int main() {
	int u, v;
	cin >> n;
	cnt = vector<int>(n, 0);
	label = vector<int>(n - 1, -1);
	for (int i = 0; i < n - 1; i++) {
		cin >> u >> v;
		edges.emplace_back(u - 1, v - 1);
		cnt[u - 1]++;
		cnt[v - 1]++;
	}
	solve();
	for (const auto &x : label)
		cout << x << endl;
	cnt.clear();
	label.clear();
}

File: since2020/CodeForces/1325C.py
'''
	@judge CodeForces
	@id 1325C
	@name Ehab and Path-etic MEXs
	@contest Codeforces Round #628

	@tag Greedy, Graph Theory
'''
from sys import stdin
from collections import Counter

def solve(n, es):
	label = [ -1 ] * (n - 1)
	c = Counter([ x for p in es for x in p ])
	m = len(list(filter(lambda x: x == 1, c.values())))
	L, R = 0, m
	for idx, uv in enumerate(es):
		u, v = uv
		if c[u] == 1 or c[v] == 1:
			label[idx] = L
			L += 1
		else:
			label[idx] = R
			R += 1
	print('\n'.join([ str(l) for l in label ]))

n  = int(input())
es = [ tuple(map(int, line.split())) for line in stdin ]
solve(n, es)